src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -63,12 +77,31 @@ trait RegularSync {
       }
   }
 
+  private def insertMinedBlock(block: Block, parentTd: BigInt) = {
+    val result: Either[BlockExecutionError, Seq[Receipt]] = ledger.executeBlock(block, blockchainStorages, validators)
+
+    result match {
+      case Right(receipts) =>
```

Comment:
is this logic different than executing blocks we receive from others? maybe we could reuse one method to handle both cases